docs/basic-features/data-fetching.md

```diff
@@ -94,6 +94,35 @@ export const getStaticProps: GetStaticProps = async context => {
 }
 ```
 
+If you want to get inferred typings for your props, you can use `StaticProps<typeof getStaticProps>` like this:
+
+```ts
+import fetch from 'node-fetch'
+import { GetStaticProps, StaticProps } from 'next'
+
+type Post = {
+  author: string
+  content: string
+}
+
+export const getStaticProps: GetStaticProps = async context => {
```

Comment:
```suggestion
export const getStaticProps = async context => {
```

Using the `GetStaticProps` type seems to break inferring. Is the utility type meant to work with the existing type like this?

```jsx
import { StaticProps, GetStaticPaths } from 'next'

type Post = {
  author: string
  content: string
}

export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: [{ params: { post: '1' } }],
    fallback: false,
  }
}

export const getStaticProps = async () => {
  const posts: Post[] = [
    {
      author: 'Vercel',
      content: 'hello wolrd',
    },
  ]

  return {
    props: {
      posts,
    },
  }
}

function Blog({ posts }: StaticProps<typeof getStaticProps>) {
  return (
    <>
      {posts.map(post => (
        <div key={post.author}>{post.author}</div>
      ))}
    </>
  )
}

export default Blog
```

<img width="927" alt="Screen Shot 2020-05-26 at 15 35 41" src="https://user-images.githubusercontent.com/22380829/82949184-c5131a00-9f68-11ea-974d-87f92a889f76.png">